vaadin-grid-flow-parent/vaadin-grid-flow/src/main/java/com/vaadin/flow/component/grid/Grid.java

```diff
@@ -3895,7 +3883,7 @@ private void generateTooltipTextData(T item, JsonObject jsonObject) {
         JsonObject tooltips = Json.createObject();
 
         idToColumnMap.forEach((id, column) -> {
-            String cellTooltip = column.getTooltipGenerator().apply(item);
+            String cellTooltip = generateTooltipContent(item, column);
             if (cellTooltip != null) {
                 tooltips.put(id, cellTooltip);
             }
```

Comment:
Looking at this again, I think we should optimize this. Currently, if you have like 50 columns, we would send the same tooltip 50 times, which can significantly increase the payload. We would also generate the tooltip 50 times, which could be a performance issue if there is database access involved.

I think we can put the grid tooltip into `tooltips` with a special key like `row`, and then in the Javascript tooltip generator below use this as a fallback if the object doesn't contain a tooltip for the specific column node id.